src/app/lend/services/lend.service.ts

```diff
@@ -0,0 +1,65 @@
+import { Injectable } from '@angular/core';
+import { BigNumber, ethers } from 'ethers';
+import { Subscription } from 'rxjs';
+import { CheddaBaseTokenVaultService } from 'src/app/contracts/chedda-base-token-vault.service';
+import { PriceOracleService } from 'src/app/contracts/price-oracle.service';
+import { environment } from 'src/environments/environment';
+import { LendingPool, LoanRequest } from '../lend.models';
+
+@Injectable({
+  providedIn: 'root'
+})
+
+export class LendService {
+
+  loanRequests: LoanRequest[] = [];
+  currentSegment = 'requests';
+  currency;
+  openLoansSubscription?: Subscription;
+  openLoanRequestsSubscription?: Subscription;
+  lendingPools: LendingPool[] = [];
```

Comment:
Have a `lendingPools` variable here that the components read the lending pool data from.
Optionally, create a `lendingPoolsSubject : BehaviorSubject<any> = new BehaviorSubject(null)`. In `loadVaultStats()` when the pools data is updated, call `this.lendingPoolsSubject.next(pools)` with the new data.
Create a subscription in the components to listen to that subject.

```
this.lendingPoolsSubscription = this.vaultStatsService.lendingPoolsSubject.subscribe(lendingPools => {
  this.lendingPools = lendingPools
}
```